scripts/commands/checkPatternsIndex.ts

```diff
@@ -0,0 +1,211 @@
+}
+
+function getExtraIndexPagesErrors(
+  indexPage: string,
+  indexEntries: Set<string>,
+  toolsEntries: Set<string>,
+): string[] {
+  return [...indexEntries]
+    .filter((page) => !toolsEntries.has(page))
+    .map(
+      (page) =>
+        `❌ ${indexPage}: The entry ${page} doesn't appear in the \`Tools\` menu.`,
+    );
+}
+
+function getExtraToolsEntriesErrors(
+  remainingToolsEntries: Set<string>,
+): string[] {
+  return [...remainingToolsEntries].map(
+    (page) => `❌ The entry ${page} is not present on any index page`,
+  );
+}
+
+function maybePrintErrorsAndFail(
+  duplicatesErrors: string[],
+  extraIndexEntriesErrors: string[],
+  extraToolsEntriesErrors: string[],
+): void {
+  let allGood = true;
+
+  if (duplicatesErrors.length > 0) {
+    duplicatesErrors.forEach((error) => console.error(error));
+    console.error(
+      `\nRemove all duplicated entries on the indices and in the Tools menu, which is set in docs/guides/_toc.json.`,
+    );
+    console.error("--------\n");
+    allGood = false;
+  }
+
+  if (extraIndexEntriesErrors.length > 0) {
+    extraIndexEntriesErrors.forEach((error) => console.error(error));
+    console.error(
+      `\nMake sure all pages have an entry in the Tools menu, which is set in docs/guides/_toc.json.`,
+    );
+    console.error("--------\n");
+    allGood = false;
+  }
+
+  if (extraToolsEntriesErrors.length > 0) {
+    extraToolsEntriesErrors.forEach((error) => console.error(error));
+    console.error(
+      "\nAdd the entries in one of the following index pages, or add the URL to the `IGNORED_URLS` list at the beginning of `/scripts/commands/checkPatternsIndex.tsx` if it's not used in Workflow:",
+    );
+    INDEX_PAGES.forEach((index) => console.error(`\t➡️  ${index}`));
+    allGood = false;
+  }
+
+  if (!allGood) {
+    process.exit(1);
+  }
+}
+
+async function main() {
+  // Todo: Remove this conditional once the migration is done. This is used only to avoid
+  // the script crashing if the file's structure doesn't exist.
+  if (!(await pathExists(TOC_PATH))) {
+    console.log(
+      `🚧 Check skipped because the migration hasn't been completed.\n`,
+    );
+    process.exit(0);
+  }
+
+  const toolsAllEntries = await getToolsTocEntriesToCheck();
+  let [toolsEntries, duplicatesErrors] = await deduplicateEntries(
+    TOC_PATH,
+    toolsAllEntries,
+  );
+
+  let extraIndexEntriesErrors: string[] = [];
+  for (const indexPage of INDEX_PAGES) {
+    const indexAllEntries = await getIndexEntries(indexPage);
+    let [indexEntries, indexDuplicatedErrors] = await deduplicateEntries(
+      indexPage,
+      indexAllEntries,
+    );
+    duplicatesErrors.push(...indexDuplicatedErrors);
+
+    extraIndexEntriesErrors.push(
+      ...getExtraIndexPagesErrors(indexPage, indexEntries, toolsEntries),
+    );
+
+    // Remove index entries from the tools entries list
+    toolsEntries = new Set(
+      [...toolsEntries].filter((page) => !indexEntries.has(page)),
+    );
```

Comment:
Taken from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/delete - this should be more clear and you could probably get rid of the comment

```suggestion
    toolsEntries.forEach((page) => {
      if (indexEntries.has(page)) {
        toolsEntries.delete(page)
       }
       }
    );
```

Reply:
Awesome, thank you Eric!